test/unit/models/discordactions.test.js

```diff
@@ -801,4 +811,184 @@ describe("discordactions", function () {
+      expect(res.totalIdle7dUsers).to.be.equal(2);
+      expect(res.totalUserRoleToBeAdded).to.be.equal(2);
+      expect(res.totalUserRoleToBeRemoved).to.be.equal(1);
+      expect(res.totalArchivedUsers).to.be.equal(1);
+    });
+  });
+
+  describe("updateGroupIdle", function () {
+    let fetchStub;
+    let allIds;
+
+    beforeEach(async function () {
+      fetchStub = fetchStub = sinon.stub(global, "fetch");
+
+      userData[0] = {
+        ...userData[0],
+        discordId: "123456789098765432",
+        discordJoinedAt: "2023-07-31T16:57:53.894000+00:00",
+        roles: { archived: false, in_discord: true },
+      };
+      userData[1] = {
+        ...userData[1],
+        discordJoinedAt: "2023-07-31T16:57:53.894000+00:00",
+        roles: { archived: false, in_discord: true },
+      };
+      userData[2] = {
+        ...userData[2],
+        discordId: "1234567",
+        discordJoinedAt: "2023-07-31T16:57:53.894000+00:00",
+        roles: { archived: false, in_discord: true },
+      };
+
+      userData[3] = {
+        ...userData[3],
+        roles: { archived: true, in_discord: false },
+      };
+
+      userData[4] = {
+        ...userData[4],
+        discordId: "2131234453456545656765767876",
+        discordJoinedAt: "2023-07-31T16:57:53.894000+00:00",
+        roles: { archived: false, in_discord: true },
+      };
+
+      getDiscordMembers[3] = {
+        ...getDiscordMembers[3],
+        roles: ["1212121212"],
+      };
+
+      await addUser(userData[0]);
+      await addUser(userData[1]);
+      await addUser(userData[2]);
+      await addUser(userData[3]);
+      await addUser(userData[4]);
+
+      getDiscordMembers[4] = {
+        ...getDiscordMembers[4],
+        roles: [...getDiscordMembers[4].roles, groupIdle.roleid, "9876543210"],
+      };
+      const addUsersPromises = userData.slice(0, 5).map((user) => userModel.add({ ...user }));
+      const responses = await Promise.all(addUsersPromises);
+      allIds = responses.map((response) => response.id);
+
+      const userStatusPromises = allIds.map(async (userId, index) => {
+        if (index === 4) {
+          await userStatusModel.updateUserStatus(userId, generateUserStatusData("ACTIVE", new Date(), new Date()));
+        } else {
+          await userStatusModel.updateUserStatus(userId, generateUserStatusData("IDLE", 1690829925336, 1690829925336));
+        }
+      });
+      await Promise.all(userStatusPromises);
+
+      const addRolesPromises = [discordRoleModel.add(groupIdle)];
+      await Promise.all(addRolesPromises);
+
+      fetchStub.returns(
+        Promise.resolve({
+          status: 200,
+          json: () => Promise.resolve(getDiscordMembers),
+        })
+      );
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+      await cleanDb();
+    });
+
+    it("should return  totalIdleUsers as 3,totalRolesApplied as 3, totalRoleToBeAdded as 3 as no one is maven", async function () {
```

Comment:
test description looks confusing to me, can we update please

Reply:
on it